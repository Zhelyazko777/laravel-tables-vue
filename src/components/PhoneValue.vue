<template>
  <a 
    :href="`tel:${value}`"
    :title="showTooltipOnHover ? tooltip : ''"
    :data-toggle="showTooltipOnHover ? 'tooltip' : ''"
  >
      {{prefix}}{{value}}{{suffix}}
  </a>
</template>

<script>
export default {
  props: {
    value: {
      required: true,
      type: String|null,
    },
    prefix: {
      required: true,
      type: String|null,
    },
    suffix: {
      required: true,
      type: String|null,
    },
    tooltip: {
      required: true,
      type: String|null,
    },
    showTooltipOnHover: {
      required: true,
      type: Boolean|null,
    },
  },
}
</script>