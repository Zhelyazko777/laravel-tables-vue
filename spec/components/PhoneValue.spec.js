import { mount } from '@vue/test-utils'
import PhoneValue from './../../src/components/PhoneValue.vue';

describe('PhoneValue.vue Test', () => {
  it('Should render corrrect value and empty suffix and preffix', () => {
    const wrapper = mount(PhoneValue, {
      propsData: {
        value: 'test value',
        prefix: null,
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();      
  });

  it('Should render corrrect value and prefix if passed and no suffix', () => {
    const wrapper = mount(PhoneValue, {
      propsData: {
        value: 'test value',
        prefix: 'prefix',
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });

    expect(wrapper).toMatchSnapshot();      
  });

  it('Should render corrrect value, suffix and prefix if passed', () => {
    const wrapper = mount(PhoneValue, {
      propsData: {
        value: 'test value',
        prefix: 'prefix',
        suffix: 'suffix',
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should render corrrect tel:', () => {
    const wrapper = mount(PhoneValue, {
      propsData: {
        value: '+359887878966',
        prefix: null,
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should render correct tooltip if passed and showTooltipOnHover is true', () => {
    const wrapper = mount(PhoneValue, {
      propsData: {
        value: null,
        prefix: null,
        suffix: null,
        tooltip: 'Some tooltip',
        showTooltipOnHover: true,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should not render correct tooltip if passed and showTooltipOnHover is false', () => {
    const wrapper = mount(PhoneValue, {
      propsData: {
        value: null,
        prefix: null,
        suffix: null,
        tooltip: 'Some tooltip',
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });
});