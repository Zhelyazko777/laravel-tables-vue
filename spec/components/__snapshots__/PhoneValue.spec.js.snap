// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneValue.vue Test Should not render correct tooltip if passed and showTooltipOnHover is false 1`] = `<a href="tel:null" title="" data-toggle=""><span class="prefix"></span> <span class="value"></span> <span class="suffix"></span></a>`;

exports[`PhoneValue.vue Test Should render correct tooltip if passed and showTooltipOnHover is true 1`] = `<a href="tel:null" title="Some tooltip" data-toggle="tooltip"><span class="prefix"></span> <span class="value"></span> <span class="suffix"></span></a>`;

exports[`PhoneValue.vue Test Should render corrrect tel: 1`] = `<a href="tel:+359887878966" title="" data-toggle=""><span class="prefix"></span> <span class="value">+359887878966</span> <span class="suffix"></span></a>`;

exports[`PhoneValue.vue Test Should render corrrect value and empty suffix and preffix 1`] = `<a href="tel:test value" title="" data-toggle=""><span class="prefix"></span> <span class="value">test value</span> <span class="suffix"></span></a>`;

exports[`PhoneValue.vue Test Should render corrrect value and prefix if passed and no suffix 1`] = `<a href="tel:test value" title="" data-toggle=""><span class="prefix">prefix</span> <span class="value">test value</span> <span class="suffix"></span></a>`;

exports[`PhoneValue.vue Test Should render corrrect value, suffix and prefix if passed 1`] = `<a href="tel:test value" title="" data-toggle=""><span class="prefix">prefix</span> <span class="value">test value</span> <span class="suffix">suffix</span></a>`;
